File: spec/lua/test_opaque_with_param.lua
local luaunit = require("luaunit")

require("opaque_with_param")

TestOpaqueWithParam = {}

function TestOpaqueWithParam:test_opaque_with_param()
    local r = OpaqueWithParam:from_file("src/term_strz.bin")

    luaunit.assertEquals(r.one.buf, "foo|b")
    luaunit.assertEquals(r.one.trailer, 0x61)
end

File: spec/python/test_opaque_with_param.py
import unittest

from opaque_with_param import OpaqueWithParam

class TestOpaqueWithParam(unittest.TestCase):
    def test_opaque_with_param(self):
        r = OpaqueWithParam.from_file("src/term_strz.bin")

        self.assertEqual(r.one.buf, "foo|b")
        self.assertEqual(r.one.trailer, 0x61)
